Lib/test/test_posix.py

```diff
@@ -22,6 +19,15 @@
 import textwrap
 from contextlib import contextmanager
 
+# Skip these tests if there is no posix or nt module.
+try:
+    import posix
+except ImportError:
+    try:
+        import nt as posix
+    except ImportError:
+        raise unittest.SkipTest("requires 'posix' or 'nt' module")
```

Comment:
Hum, when is this code path supposed to be taken? On Python implementations other than CPython? If the compilation of the posix/nt extension failed?

I would prefer to get an ImportError rather than skipping silently the test. If there is a legit use case to skip the test, in that ok, sure, we can skip it. But now it's unclear to me why posix and nt would miss.

PyPy has a `posix` module on Linux for example.

Reply:
I do not know, but maybe on some custom builds of Python on some exotic platforms.

Currently the os module only supports posix and nt.

Reply:
> I do not know, but maybe on some custom builds of Python on some exotic platforms.

If someone has a *custom* build of Python, I suppose that they can modify one line of test_posix.py to skip the whole file on their *exotic* platform. I would prefer to be safe by default. For me, it's more likely that the test would be skipped by mistake, than causing troubles on exotic platforms. Also, we don't support exotic platforms (PEP 11).